code/datums/elements/decals/blood.dm

```diff
@@ -27,6 +27,11 @@
 	SIGNAL_HANDLER
 
 	var/atom/A = source
+	var/list/blood_dna = GET_ATOM_BLOOD_DNA(A)
+	if(!blood_dna)
+		return
+
+	var/blood_color = get_blood_dna_color(blood_dna)
```

Comment:
This needs to be able to adapt to low contrasts automatically and handle any colour of blood with either light/dark mode. To do this, put the colour into HSL, if the lightness value is either below 30 or above 70 (depending on dark/light mode), then simply 100-lightness the colour and form a new colour from that. Dark reds will become a light pink, white will become black, etc.

We have no idea what coloured bloods will be added in the future, so we need to be prepared for that.